code/modules/mob/living/simple_animal/abnormality/he/red_shoes.dm

```diff
@@ -0,0 +1,409 @@
+		del_on_death = TRUE
+	density = FALSE
+	for(var/obj/O in src)
+		O.forceMove(loc)
+	if(possessee)//hopefully this refers to the specific individual
+		var/mob/living/carbon/human/H = possessee
+		possessee.status_flags &= ~GODMODE
+		possessee.forceMove(loc)
+		possessee = null
+		H.adjustBruteLoss(500)//the host dies
+	for(var/mob/living/carbon/human/H in GLOB.mob_living_list)//stops possessing people, prevents runtimes. Panicked players are ghosted so use mob_living_list
+		UnPossess(H)
+	say(pick(death_lines))
+	alpha = 255
+	QDEL_IN(src, 10 SECONDS)
+	QDEL_NULL(soundloop)
+	..()
+
+/mob/living/simple_animal/hostile/abnormality/red_shoes/Initialize()
+	. = ..()
+	if(locate(/obj/structure/redshoes_cushion) in get_turf(src))
+		icon_state = "redshoes"
+		update_icon()
+	soundloop = new(list(src), FALSE)
+
+/mob/living/simple_animal/hostile/abnormality/red_shoes/PostSpawn()
+	if(locate(/obj/structure/redshoes_cushion) in get_turf(src))
+		return
+	new /obj/structure/redshoes_cushion(get_turf(src))
+	soundloop.start()
+	return ..()
+
+//*** Work Mechanics ***//
+/mob/living/simple_animal/hostile/abnormality/red_shoes/proc/Apply_Desire(mob/living/carbon/human/user)
+	user.apply_status_effect(STATUS_EFFECT_REDPOSSESS)//desire applied
+	playsound(src, 'sound/abnormalities/redshoes/RedShoes_Activate.ogg', 100, 1)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/red_shoes/FailureEffect(mob/living/carbon/human/user, work_type, pe)
+	datum_reference.qliphoth_change(-1)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/red_shoes/PostWorkEffect(mob/living/carbon/human/user, work_type, pe)
+	if(get_attribute_level(user, TEMPERANCE_ATTRIBUTE) < 60)
+		Apply_Desire(user)
+		user.adjustSanityLoss(500)
+		user.visible_message("<span class='userdanger'>[user] ignores [p_their()] orders and continually glances at The Red Shoes. Now [p_theyre()] reaching out their hand to take the shoes.</span>", "<span class='userdanger'>What lovely shoes...</span>")
+
+//***Breach Mechanics***//
+/mob/living/simple_animal/hostile/abnormality/red_shoes/ZeroQliphoth(mob/living/carbon/human/user)//silent girl with extra steps
+	if(!(status_flags & GODMODE)) // If it's breaching right now
+		return
+	if(possessee)//If the first check fails
+		return
+	SLEEP_CHECK_DEATH(30)
+	if(LAZYLEN(GLOB.player_list) < 3)//solo breach if there aren't many players
+		BreachEffect()
+		return
+	numbermarked = (1 + round(LAZYLEN(GLOB.player_list) / 6))
+	var/list/potentialmarked = list()//mark code from cherry
+	var/list/marked = list()
+	for(var/mob/living/carbon/human/L in GLOB.player_list)
+		if(L.stat >= HARD_CRIT || L.sanity_lost || z != L.z) // Dead or in hard crit, insane, or on a different Z level.
+			continue
+		potentialmarked += L
+	for(var/i=numbermarked, i>=1, i--)
+		var/mob/living/Y = pick(potentialmarked)
+		if(faction_check_mob(Y, FALSE) || Y.z != z || Y.stat == DEAD)
+			continue
+		if(Y in marked)
+			continue
+		marked+=Y
+		Apply_Desire(Y)
```

Comment:
```suggestion
	var/mob/living/Y
	while(numbermarked > marked.len && potentialmarked.len > 0)
		Y = pick(potentialmarked)
		potentialmarked -= Y
		if(faction_check_mob(Y, FALSE) || Y.z != z || Y.stat == DEAD)
			continue
		marked+=Y
		Apply_Desire(Y)
```
A better way to write this that results in less processing time due to repeatedly picking invalid targets.